src/acsets/ColumnImplementations.jl

```diff
@@ -0,0 +1,592 @@
+  v::Vector{T}
+  function SentinelVecMap{T,D}() where {T,D}
+    new{T,D}(Vector{T}())
+  end
+  function SentinelVecMap{T,D}(v::Vector{T}) where {T,D}
+    new{T,D}(v)
+  end
+end
+
+"""
+In a FilledVecMap, the defined indices are equal to the definable indices. When
+the definable indices are expanded, they are filled with the default value, and
+thus the defined indices are also expanded.
+"""
+@struct_hash_equal struct FilledVecMap{T,D<:Default{T}} <: DefaultVecMap{T,D}
+  v::Vector{T}
+  function FilledVecMap{T,D}() where {T,D}
+    new{T,D}(Vector{T}())
+  end
+  function FilledVecMap{T,D}(v::Vector{T}) where {T,D}
+    new{T,D}(v)
+  end
+end
+
+Base.getindex(m::DefaultVecMap, x) = m.v[x]
+
+function Base.setindex!(m::DefaultVecMap, y, x::Int)
+  m.v[x] = y
+end
+
+function Base.view(m::DefaultVecMap, xs)
+  view(m.v, xs)
+end
+
+
+Base.copy(m::M) where {T,D,M<:DefaultVecMap{T,D}} = M(copy(m.v))
+
+# This only works if f::T->T
+Base.map(f, m::M) where {T,D,M<:DefaultVecMap{T,D}} = M(map(f, m.v))
+
+Columns.isdefined(m::SentinelVecMap{T,D}, x::Int) where {T,D} = m.v[x] != default(D)
+
+Columns.isdefined(m::FilledVecMap{T,D}, x::Int) where {T,D} = true
+
+Columns.isdefinable(m::DefaultVecMap, x::Int) = x ∈ 1:length(m.v)
+
+function Columns.undefine_or_clear!(m::DefaultVecMap{T,D}, x::Int) where {T,D}
+  m.v[x] = default(D)
+end
+
+function Columns.set_definable!(m::DefaultVecMap{T,D}, xs::Base.OneTo{Int}) where {T,D}
+  n = length(m.v)
+  resize!(m.v, xs.stop)
+  for i in (n+1):xs.stop
+    m.v[i] = default(D)
+  end
+end
+
+Columns.isstored(m, x) = true
+Columns.store!(m, x) = nothing
+
+"""
+This is a dictionary-backed mapping, that is partial.
+"""
+@struct_hash_equal struct PartialDictMap{S,T} <: PartialMapping{S,T}
+  d::Dict{S,T}
+  function PartialDictMap{S,T}(pairs) where {S,T}
+    new{S,T}(Dict{S,T}(pairs))
+  end
+  function PartialDictMap{S,T}() where {S,T}
+    new{S,T}(Dict{S,T}())
+  end
+end
+
+Base.getindex(m::PartialDictMap, xs) =
+  broadcast(xs) do x
+    m.d[x]
+  end
+
+function Base.setindex!(m::PartialDictMap{S}, y, x::S) where {S}
+  m.d[x] = y
+end
+
+Base.view(m::PartialDictMap, xs) = view(m.d, xs)
+
+Base.copy(m::PartialDictMap{S,T}) where {S,T} = PartialDictMap{S,T}(copy(m.d))
+
+# Note: this only works if f::T->T
+Base.map(f, m::PartialDictMap{S,T}) where {S,T} =
+  PartialDictMap{S,T}([(k,f(v)) for (k,v) in m.d])
+
+Columns.isdefined(m::PartialDictMap{S}, x::S) where {S} = x ∈ keys(m.d)
+Columns.undefine_or_clear!(m::PartialDictMap{S}, x::S) where {S} = delete!(m.d, x)
+
+Columns.isdefinable(m::PartialDictMap{S}, x::S) where {S} = true
+
+Columns.set_definable!(m::PartialDictMap, x) = nothing
+
+Columns.isstored(m::PartialDictMap, x) = true
+Columns.store!(m::PartialDictMap, x) = nothing
+
+"""
+This is a dictionary-backed mapping that returns a default value
+when accessed at an undefined index.
+"""
+@struct_hash_equal struct DefaultDictMap{S,T,D<:Default{T}} <: DefaultMapping{S,T}
+  d::Dict{S,T}
```

Comment:
Would it simplify the code to use `DefaultDict` from DataStructures (which we already depend on)?

Reply:
No, because this is done in a consistent way with DefaultVecMap, and if I used DefaultDict then this would be different.